tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py

```diff
@@ -0,0 +1,270 @@
+                                 'with the inferred problem type. '
+                                 'Inferred problem type={}, predicted proba shape={}'.format(
+                    self.problem_type, y_pred_proba.shape))
+            else:
+                return y_pred_proba
+        else:
+            return y_pred_proba
+
+    def _get_default_auxiliary_params(self) -> dict:
+        default_auxiliary_params = super()._get_default_auxiliary_params()
+        extra_auxiliary_params = dict(
+            get_feature_kwargs=dict(
+                ignored_type_group_raw=[R_CATEGORY],
+                ignored_type_group_special=[S_TEXT_NGRAM, S_TEXT_SPECIAL, S_TEXT_AS_CATEGORY],
+            )
+        )
+        default_auxiliary_params.update(extra_auxiliary_params)
+        return default_auxiliary_params
+
+    def _set_default_params(self):
+        try:
+            from autogluon.text.text_prediction.dataset import TabularDataset
+            from autogluon.text.text_prediction.text_prediction import ag_text_prediction_params
+        except ImportError:
+            raise ImportError(AG_TEXT_IMPORT_ERROR)
+        self.params = ag_text_prediction_params.create('default')
+
+    def _fit(self,
+             X_train: pd.DataFrame, y_train: pd.Series,
+             X_val: Optional[pd.DataFrame] = None,
+             y_val: Optional[pd.Series] = None,
+             time_limit: Optional[int] = None, **kwargs):
+        """The internal fit function
+
+        Parameters
+        ----------
+        X_train
+            Features of the training dataset
+        y_train
+            Labels of the training dataset
+        X_val
+            Features of the validation dataset
+        y_val
+            Labels of the validation dataset
+        time_limit
+            The time limits for the fit function
+        kwargs
+            Other keyword arguments
+
+        """
+        try:
+            from autogluon.text.text_prediction.dataset import TabularDataset
+        except ImportError:
+            raise ImportError(AG_TEXT_IMPORT_ERROR)
+
+        # Get arguments from kwargs
+        verbosity = kwargs.get('verbosity', 2)
+        num_cpus = kwargs.get('num_cpus')
+        num_gpus = kwargs.get('num_gpus')
+        X_train = X_train[['sentence_raw_text']]
+        X_val = X_val[['sentence_raw_text']]
+        column_properties = self._build_model(X_train=X_train,
+                                              y_train=y_train,
+                                              X_val=X_val,
+                                              y_val=y_val,
+                                              hyperparameters=self.params)
+        print('X_train=', X_train)
+        # Insert the label column
+        X_train.insert(len(X_train.columns), self._label_column_name, y_train)
+        X_val.insert(len(X_val.columns), self._label_column_name, y_val)
+        scheduler_options = self.params['hpo_params']['scheduler_options']
```

Comment:
why is hyperparameter tuning logic in `_fit`? This should probably not be present. First lets get it working without HPO and then we can add HPO in properly (reference other model implementations for an idea of what is needed)

Reply:
This comment still applies.